### DISCUSSION TITLE
Problem with root and problem with the grabber

### DISCUSSION THREAD

**Discussion by LeSenegal** [awawa-dev/HyperHDR#834] [Q&A] [answered]
Hello,
I had hyperhdr installed in version 20.0.0.0, but I changed my internet box, so I reinstalled it, but this time I can't get it to work. I tested pi os in 32 bits, 64 bits, reinstalling hyperhdr several times. I've also tried using the hyperhdr image, but it doesn't work. I can access to the dashboard but led didn't work. 
In the log that's said : 
[LEDDEVICE0_WS281X] Device 'ws281x' is disabled due to an error: 'Failed to open. Error message: mmap() failed. YOUR NEED TO RUN HYPERHDR AS A ROOT USER FOR MMAP TO WORK. SUCH CONFIGURATION IS NOT OFFICIALLY SUPPORTED.'

I already tested with : 
sudo systemctl stop hyperhdr@pi.service
sudo systemctl disable hyperhdr@pi
sudo systemctl enable hyperhdr@root

And I now have a problem with my grabber that inverse the colors I think : 
Image captured by the grabber : 
![shot211059](https://github.com/awawa-dev/HyperHDR/assets/142126867/b3fad5dc-cb3e-4b59-b972-2df6aacbadb2)
Real image :
![shot211060](https://github.com/awawa-dev/HyperHDR/assets/142126867/bf29d572-1d57-48ba-b3c1-91a155ba934c)

My configuration actually : 
- Raspberry Pi 3 Model B Plus Rev 1.3
- led WS2812B
- Navy U3 grabber
- HyperHDR 20.0.0.0
- And the OS :
![image](https://github.com/awawa-dev/HyperHDR/assets/142126867/b8512d54-02be-4ccd-b036-d44d68a11d35)

Thank you for your help

**Comment by awawa-dev** [comment, marked as answer]
Hi
You may be using a different user than `pi`, which only exists in our images, because the official flasher and images have removed it and you can use a non-standard username (then you need to adapt the command for systemctl, you need to stop the service for the user it is currently installed and running on ). BTW `sudo systemctl enable hyperhdr@root` doesnt start the service and the restart is required. As for inverted colors it's a kernel bug: https://github.com/awawa-dev/HyperHDR/discussions/797

**Comment by LeSenegal** [comment]
Thank you I try for the led problem, that works fine, and for the grabber problem where can I find an older image of RaspiOS ?
Or should I get the Raspberry Pi OS version Legacy ?

**Comment by awawa-dev** [comment]
I think HyperHDR OS images have an older kernel version because they are based on Debian Bullseye and not the latest Bookworm. Or you can also manually downgrade the Bookworm kernel using the standard Debian procedure.

**Comment by LeSenegal** [comment]
Effectively, the HyperHDR OS image  is on kernel 6.1.21, all works, thanks for your time. Love your work

**Reply by drharis** [reply]
Hi there,
I have same issue everything works, but colours are different. I think it is negative than inverted, I am not sure. Please see image. Do I have to downgrade the kernal version? If so, I would be grateful if you could let me know how to do it. I am using a Raspberry pi 2B, connected via ethernet. Firestick plugged into HDMI port of the capture card and output to my projector. USB to raspberry pi. Using hyperHDR 20.0
Many thanks. Spent hours on this trying to fiddle with RGB, BRG etc but no luck
![IMG_20240510_131525](https://github.com/awawa-dev/HyperHDR/assets/33908724/80e217de-559e-4c4e-bbdf-830681e003ae)

**Reply by awawa-dev** [reply]
Hi
@drharis have you noticed and read the pinned post in the discussion section?

**Reply by drharis** [reply]
Thanks for the quick response. If you are referring to https://github.com/awawa-dev/HyperHDR/discussions/797 , yes I read it. It says kernerl ptoblem and needs to downgrade. I was asking if that would fix the issue, someone could guide me how to do it. Do I go to the legacy OS and flash with the one with kernal version 6.1 and re-install hyperHDR?

**Reply by awawa-dev** [reply]
so far nobody has provided such manual and I'm still using Bullseye so dont have this issue.
And don't you want to experiment and upgrade the kernel instead using mentioned `sudo rpi-update` command?

https://www.raspberrypi.com/documentation/computers/os.html#using-rpi-update

**Reply by awawa-dev** [reply]
> Thanks for the quick response. If you are referring to https://github.com/awawa-dev/HyperHDR/discussions/797

no, this topic which is pinned at the top in the discussion section:
https://github.com/awawa-dev/HyperHDR/discussions/848

**Reply by drharis** [reply]
Thanks. The sudo update command only updates to the latest kernel and OS as per the documentation you linked. I installed the lite OS on fresh SD card last night. So I presume it will be the latest kernel and OS. I'm not sure if we can downgrade using those commands.

**Reply by awawa-dev** [reply]
You did not read or understand the instructions provided or the related discussion on the picamera2 forum that explained what this command does: provide a pre-production beta kernel that is not available anywhere else. May contain other bugs, but the released "stable" kernel also contains a critical colors-affecting bug. And according to picamera2 this beta kernel solved a similar problem.

Since this is a kernel bug and not a problem of our application, I think we have covered the topic with available solutions, the rest is up to the user to share his experience and provide feedback to others. Or if you want the simplest solution, you should probably wait a while for the Raspberry Pi kernel channel until the kernel release, which is currently in beta testing.

**Reply by drharis** [reply]
Hi, 
You are right, I did not understand exactly the instruction on that webpage. Thanks for clarifying. Now, I that I have some more time to play around with it, I can say that it is working fine now. I tried the sudo update command and the kernel did get updated but the problem persisted. So, I removed the card and re-flashed with legacy Bullseye version on pi OS and reinstalled HyperHDR. It all worked fine! Thanks for your help. If it was not for that I would have been spending many more hours and pulling whatever hair left in my head! Thanks again

**Reply by awawa-dev** [reply]
It's great that you managed to solve the problem. I just have a question because I want to make sure which command you used before and which one didn't help, because it will also be information for others.
`sudo update` (there is no such command I think) or `sudo apt update` or `sudo apt upgrade` or `sudo rpi-update`
Did you restart Rpi after that?

**Reply by drharis** [reply]
I tried sudo rpi-update as per the pi camera website link you provided

**Reply by awawa-dev** [reply]
Ok, thanks. So the source of the RPi kernel/firmware problem is still unknown and unfortunately we don't know if someone is working on it or is aware of it. I guess it's limited to RPi only, we havent received similar reports from x86_x64 systems yet.
